### ISSUE TITLE
[clang-tidy] modernize-use-using suggested for extern "C" typedefs

### ISSUE THREAD

**Issue by LegalizeAdulthood** [llvm/llvm-project#141394]
Repro: https://godbolt.org/z/6jdqMrjKs

The following code:
```
extern "C"
{
typedef int int_t;
}
```
should not issue any warnings with `modernize-use-using` because the typedef is inside an `extern "C"` block.

**Comment by frederick-vs-ja** [comment]
IIUC the example is still putting the `typedef` declaration in `.cpp` source file. Can we show an example where the `typedef` declaration is put into a header (available in both C and C++) and the header is included by a C++ source file?

**Comment by localspook** [comment]
+1 on `IgnoreExternC`, to me it seems exactly what's being asked for. I think that resolves this issue, but feel free to reopen if there's more to say.

**Comment by localspook** [comment]
I guess it could make sense if you have a pure C++ project and are using `extern "C"` not to make your code compilable as C,  but just to provide a stable ABI. I could see an argument for changing `IgnoreExternC` to be on by default (I even found a [comment to that effect](https://github.com/llvm/llvm-project/pull/180326#pullrequestreview-3771085553)).

**Comment by voyager-jhk** [comment]
hi, I've opened pr #199189 to resolve this by changing the default of `IgnoreExternC` to `true`.
